src/lib/y2packager/product_location.rb

```diff
@@ -10,22 +10,95 @@
 # FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
 # ------------------------------------------------------------------------------
 
+require "yast"
+
+require "y2packager/repomd_downloader"
+require "y2packager/solvable_pool"
+require "y2packager/product_finder"
+
+Yast.import "URL"
+
 module Y2Packager
   # This class represents a product located on a multi-repository medium,
   # libzypp reads the available products from /medium.1/products file.
   class ProductLocation
-    # @return [String] Products on the medium
+    include Yast::Logger
+
+    # @return [String] Product name (user name from the /medium.1/products file)
     attr_reader :name
-    # @return [String] User selected products
+    # @return [String] Path on the medium (relative to the medium root)
     attr_reader :dir
 
+    # @return [String] Product name (the zypp resolvable)
+    attr_reader :product_name
+    # @return [String] The product package (*-release RPM usually)
+    attr_reader :product_package
+    # @return [String] Product summary
+    attr_reader :summary
+    # @return [String] Product description
+    attr_reader :description
+    # @return [Integer] Display order
```

Comment:
from code it looks like it can be also nil. What happens in that case?

Reply:
It's not used yet, but later the product sorter already allows `nil`, it's OK.